pkg/infrastructure/azure/azure.go

```diff
@@ -122,20 +122,26 @@ func (p *Provider) PreProvision(ctx context.Context, in clusterapi.PreProvisionI
 	if err != nil {
 		return fmt.Errorf("failed to get azure resource groups factory: %w", err)
 	}
+	// Check if the resource group exists. Create or update function replaces metadata
+	// like existing tags if the resource group exists.
```

Comment:
We should not replace tags on the Resource Groups, we can append but we shouldn't do anything that remove's user tags. Once the ARO cluster is down our deny assignment will prevent new tags and this can break customer security policies if we remove tags then prevent new tag creation. ARO and Customer tags must co-locate.

Reply:
Yeah I should probably write a more clear comment. This piece of code prevents existing tag manipulation

Reply:
Great thanks!